pkg/controller/servicebinding/binder.go

```diff
@@ -195,16 +196,20 @@ func (b *binder) updateVolumes(volumes []interface{}) ([]interface{}, error) {
 }
 
 // removeVolumes remove the bind volumes from informed list of unstructured volumes.
-func (b *binder) removeVolumes(volumes []interface{}) []interface{} {
+func (b *binder) removeVolumes(volumes []interface{}) ([]interface{}, error) {
 	name := b.sbr.GetName()
 	var cleanVolumes []interface{}
 	for _, v := range volumes {
-		volume := v.(corev1.Volume)
-		if name != volume.Name {
+		volume, ok := v.(map[string]interface{})
```

Comment:
@isutton 

> Why has been the type changed in here?

The `unbind(...)` method calls the `removeVolumes(...)`. This part was not tested before. This issue is caught through my PR.

> We should add stronger types whenever required, but try to not do the other way around.

We need just the name of the volume in this removeVolume(...) function.

Also, in `updateVolume`, we're converting volume to map[string]interface{} -> https://github.com/redhat-developer/service-binding-operator/blob/master/pkg/controller/servicebinding/binder.go#L169

Reply:
I believe `updateVolume`, `removeVolumes` and `extractSpecVolumes` could be changed to use `corev1.Volume` instead; this has the added benefit of being easier to access members and reason about what is inside.

Reply:
Created an issue here https://github.com/redhat-developer/service-binding-operator/issues/821

Reply:
This issue came up because this part was not tested before. We can address this in another PR.

Reply:
> This issue came up because this part was not tested before. We can address this in another PR.

@DhritiShikhar what issue are you referring to? It is unclear to me, hence the questions above.